File: c/p007.c
/* Find the 10001st prime number. */

#include "utils.c"
#include <stdio.h>

#define PRIMES 10001

int main(char * args[]) {
	int primes[PRIMES] = {0};
	int test = 0;
	int i = 0;
	for (i = 0; i < PRIMES; i++) {
		while (!is_prime(test)) {
			test++;	
		}
		primes[i] = test;
		test++;
	}	


	printf("%d\n", primes[PRIMES]);
	return 0;
}

File: python/p007.py
# Find the 10001st prime

from utils import is_prime
import sys


TARGET = 10001

# Start at 1 so we can test only odd numbers
count = 1
no = 1

while count < TARGET:
    no += 2
    if (is_prime(no)):
        count += 1

print no


